wa/framework/workload.py

```diff
@@ -85,53 +93,117 @@ def __str__(self):
         return '<Workload {}>'.format(self.name)
 
 
+class ApkUiautoWorkload(Workload):
+    
+    platform = 'android'
+
+    def __init__(self, target, **kwargs):
+        super(ApkUiautoWorkload, self).__init__(target, **kwargs)
+        self.apk = ApkHander(self)
+        self.gui = UiAutomatorGUI(self)
+
+    def init_resources(self, context):
+        self.apk.init_resources(context.resolver)
+        self.gui.init_resources(context.resolver)
+        self.gui.init_commands()
+
+    def initialize(self, context):
+        self.gui.deploy()
+
+    def setup(self, context):
+        self.apk.setup(context)
+        self.gui.setup()
+
+    def run(self, context):
+        self.gui.run()
+
+    def extract_results(self, context):
+        self.gui.extract_results()
+
+    def teardown(self, context):
+        self.gui.teardown()
+        self.apk.teardown()
+
+    def finalize(self, context):
+        self.gui.remove()
+
+
 class UiAutomatorGUI(object):
 
-    def __init__(self, target, package='', klass='UiAutomation',
-                 method='runUiAutoamtion'):
-        self.target = target
+    stages = ['setup', 'runWorkload', 'extractResults', 'teardown']
+
+    def __init__(self, owner, package=None, klass='UiAutomation', timeout=600):
+        self.owner = owner
+        self.target = self.owner.target
         self.uiauto_package = package
         self.uiauto_class = klass
```

Comment:
Yes. `class` is a reversed word in Python. So if you want to use it as an identifier, you use `klass` instead (some prefer `class_` instead, but I'm not a fan of trailing underscores).